ipc_receivefile.c

```diff
@@ -0,0 +1,400 @@
+////////////////////////////////////////////////////////////////////////////////
+//IPC RECEIVER
+//
+//
+//
+////////////////////////////////////////////////////////////////////////////////
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <errno.h>
+#include <sys/iofunc.h>
+#include <sys/dispatch.h>
+#include <sys/neutrino.h>
+
+#include "ipc_common.h"
+
+#include <getopt.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+
+#include <sys/mman.h>
+#include <process.h>
+#include <string.h>
+#include <unistd.h>
+
+typedef union
+{
+	uint16_t msg_type;
+	struct _pulse pulse;
+	cksum_header_t cksum_hdr;
+} msg_buf_t;
+
+typedef union
+{
+	uint16_t type;
+	struct _pulse pulse;
+	get_shmem_msg_t get_shmem;
+	changed_shmem_msg_t changed_shmem;
+} recv_buf_t;
+
+void ipcMessagePassingReceive(int fd);
+int create_shared_memory(unsigned nbytes, int client_pid, void **ptr, shm_handle_t *handle);
+void ipcSharedMemoryReceive(int fd);
+void ipcPipeReceive(int fd);
+
+int main(int argc, char* argv[])
+{
+    int fd = -1;
+    int longindex;
+    int ch;
+    char method;
+    static struct option longopts[] = {
+        { "help", no_argument,  NULL, 'h' },
+		{ "messages", no_argument,  NULL, 'm' },
+        { "queue", no_argument,  NULL, 'q' },
+        { "pipe",   no_argument, NULL, 'p' },
+        { "shm",   no_argument, NULL, 's' },
+        { "file",   required_argument, NULL, 'f' },
+        { NULL, 0, NULL, 0 }
+    };
+
+    while ((ch = getopt_long(argc, argv, "hmqpsf:", longopts, &longindex)) != -1) {
+
+        switch (ch) {
+        case 'f':
+                if ((fd = open(optarg, O_WRONLY)) == -1)
+                	printf("      Unable to open the file '%s'\n", optarg);
+                //the target file chosen by receiver should be empty
+                if(getFileSize(fd)!=0)
+                	printf("The file should be empty!");
+                break;
+        case 'm':
+        		if(method!=0){

```

Comment:
method is a char 0  is an int. And you need to initialize method before testing it.

![image](https://user-images.githubusercontent.com/97879298/155983113-e9f0a805-ae22-4bcd-a3db-a81f9732c270.png)

Reply:
All relevant things have been modified. Could you check it now?